File: minion-app/src/app/models/entities/Edge.ts
import {from} from "rxjs";

export class Edge {

  id: number;
  fromID: number;
  toID: number;
  color: string;

  constructor(id?: number, fromID?: number, toID?: number, color?: string) {
    this.id = id;
    this.fromID = fromID;
    this.toID = toID;
    this.color = color;
  }

}

export class VisEdge {

  id: number;
  from: number;
  to: number;
  color: string;

  constructor(id?: number, from?: number, to?: number, color?: string) {
    this.id = id;
    this.from = from;
    this.to = to;
    this.color = color;
  }

}

File: java-engine/src/main/java/hu/elteik/knowledgelab/javaengine/core/models/base/Edge.java
package hu.elteik.knowledgelab.javaengine.core.models.base;

import hu.elteik.knowledgelab.javaengine.core.models.base.Color;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Edge {

    private Long ID;
    private Long fromID;
    private Long toID;
    private Color color;

}
